README.md

```diff
@@ -1150,6 +1158,145 @@ the opposite order:
+```
+
+If a table of the schema type name does not exist, an exception will be raised at compile time. See
+[Configuring the SQL Database to Match the GraphQL Schema](#configuring-the-sql-database-to-match-the-graphql-schema)
+for a possible option to resolve such naming discrepancies.
+
+
+### End-To-End SQL Example
+An end-to-end example including relevant GraphQL schema and SQLAlchemy engine preparation follows.
+
+This is intended to show the setup steps for the SQL backend of the GraphQL compiler, and
+does not represent best practices for configuring and running SQLAlchemy in a production system. 
+
+```python
+from graphql import parse
+from graphql.utils.build_ast_schema import build_ast_schema
+from sqlalchemy import MetaData, Table, Column, String, create_engine
+from graphql_compiler.compiler.ir_lowering_sql.metadata import SqlMetadata
+from graphql_compiler import compile_graphql_to_sql, insert_arguments_into_query
+
+# Step 1: Configure a GraphQL schema (note that this can also be done programmatically)
+schema_text = '''
+schema {
+    query: RootSchemaQuery
+}
+# IMPORTANT NOTE: all compiler directives are expected here, but not shown to keep the example brief
+
+directive @filter(op_name: String!, value: [String!]!) on FIELD | INLINE_FRAGMENT
+
+# < ... more directives here ... >
+
+directive @output(out_name: String!) on FIELD
+
+type Animal {
+    name: String
+}
+'''
+schema = build_ast_schema(parse(schema_text))
+
+# Step 2: For all GraphQL types, bind all corresponding SQLAlchemy Tables to a single SQLAlchemy
+# metadata instance, using the expected naming detailed above.
+# See https://docs.sqlalchemy.org/en/latest/core/metadata.html for more details on this step.
+metadata = MetaData()
+animal_table = Table(
+    'animal', # name of table matches type name from schema
+    metadata,
+    # Animal.name schema field has corresponding 'name' column in animal table
+    Column('name', String(length=12)), 
+)
+
+# Step 3: Prepare a SQLAlchemy engine to query the target relational database.
+# See https://docs.sqlalchemy.org/en/latest/core/engines.html for more detail on this step.
+engine = create_engine('<connection string>')
+
+# Step 4: Wrap the SQLAlchemy metadata and dialect as a SqlMetadata GraphQL compiler object
+sql_metadata = SqlMetadata(engine.dialect, metadata)
+
+# Step 5: Prepare and compile a GraphQL query against the schema
+graphql_query = '''
+{
+    Animal {
+        name @output(out_name: "animal_name")
+             @filter(op_name: "in_collection", value: ["$names"])
+    }
+}
+'''
+parameters = {
+    'names': ['animal name 1', 'animal name 2'],
+}
+
+compilation_result = compile_graphql_to_sql(schema, graphql_query, sql_metadata)
+compilation_result = insert_arguments_into_query(compilation_result, parameters)
+
+# Step 6: Execute compiled query against a SQLAlchemy engine/connection. 
+# See https://docs.sqlalchemy.org/en/latest/core/connections.html for more details.
+query = compilation_result.query
```

Comment:
Another thing - here `compilation_result` should be a string (since it is the result of `insert_arguments_into_query`) and would not have an attribute `query`. I think you should just be able to `execute(compilation_result)`